app/facades/websocket/WebSocketController.scala

```diff
@@ -174,14 +199,37 @@ class WebSocketController @Inject() (actorSystem:  ActorSystem,
       Future.successful(Left[Result, DSAFlow](Forbidden))
   }
 
-  private def acceptOrResult(f: Option[DSLinkSessionInfo] => Future[Either[Result, Flow[DSAMessage, DSAMessage, _]]]): WebSocket = {
+  private def acceptOrResult(f: Option[DSLinkSessionInfo] => Future[Either[Result, DSAFlow]]): WebSocket = {
     WebSocket { request =>
       log.debug(s"WS request received: $request")
       val dsId = getDsId(request)
+      val clientAuth = getAuth(request)
       val sessionInfo = cache.get[DSLinkSessionInfo](dsId)
       log.debug(s"Session info retrieved for $dsId: $sessionInfo")
 
-      f(sessionInfo).map(_.right.map(getTransformer(sessionInfo).transform))
+      val res =
+        if (validateAuth(sessionInfo, clientAuth))
+          f(sessionInfo)
+        else {
+          log.warn(s"Authentication failed, auth value is not correct.")
+          val failedResult = Result(new ResponseHeader(UNAUTHORIZED
+                                                       , reasonPhrase = Option(s"Connection failed: DSLink auth is wrong: '" +
+                                                                               clientAuth + "'")
+                                                      )
+                                    , HttpEntity.NoEntity
+                                   )
+          Future.successful(Left[Result, DSAFlow](failedResult))
+        }
+      
+      res.map(_.right.map(getTransformer(sessionInfo).transform))
+    }
+  }
+
+  private def validateAuth(si: Option[DSLinkSessionInfo], clientAuth: Option[String]): Boolean = {
+    val ci = si.getOrElse(return false).ci
+    val localAuth = LocalKeys.saltSharedSecret(ci.salt.getBytes, ci.sharedSecret)
+    clientAuth.fold(false) { v =>
+      localAuth == v
     }
```

Comment:
Is there some difference with 
`clientAuth.getOrElse() == localAuth `
instead of 
```
clientAuth.fold(false) { v =>
   localAuth == v
}
```
?

Reply:
Yes. It is the same. Changed.